server/api/src/rsDataObjOpen.cpp

```diff
@@ -58,15 +63,86 @@
 #include "irods_server_api_call.hpp"
 #include "irods_at_scope_exit.hpp"
 #include "key_value_proxy.hpp"
+#include "replica_access_table.hpp"
+
+#define RODS_SERVER
+#include "irods_query.hpp"
+#undef RODS_SERVER
 
 #include "boost/format.hpp"
 
 #include "irods_logger.hpp"
+#include <stdexcept>
+
+#define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
+#include "filesystem.hpp"
+
+#include <fmt/format.h>
+
+#include <sys/types.h>
+#include <unistd.h>
 
 using logger = irods::experimental::log;
 
 namespace {
 
+enum class update_operation
+{
+    create,
+    update
+};
```

Comment:
this feels too generic. maybe use better names? `update_operation` tells us nothing about what it relates to.

Reply:
add a comment, so this is fine.